qcodes/dataset/database_copy_paste.py

```diff
@@ -0,0 +1,397 @@
+
+    # Fetch the attributes of the runs' experiment
+    # hopefully, this is enough to uniquely identify the experiment
+
+    exp_attr_names = ['name', 'sample_name', 'start_time', 'end_time',
+                      'format_string']
+
+    attrs_query = f"""
+                  SELECT {','.join(exp_attr_names)}
+                  FROM experiments
+                  WHERE exp_id = ?
+                  """
+    cursor = source_conn.cursor()
+    cursor.execute(attrs_query, (source_exp_ids[0],))
+    row = cursor.fetchall()[0]
+    exp_attrs = {attr: row[attr] for attr in exp_attr_names}
+
+    # Massage the target DB file to accomodate the runs
+    # (create new experiment if needed)
+
+    target_conn = connect(target_db_path)
+
+    # this function raises if the target DB file has several experiments
+    # matching both the name and sample_name
+
+    try:
+        with atomic(target_conn) as target_conn:
+
+            target_exp_id = _create_exp_if_needed(target_conn,
+                                                  exp_attrs['name'],
+                                                  exp_attrs['sample_name'],
+                                                  exp_attrs['format_string'],
+                                                  exp_attrs['start_time'],
+                                                  exp_attrs['end_time'])
+
+            # Finally insert the runs
+            for run_id in run_ids:
+                _copy_single_dataset_into_db(DataSet(run_id=run_id,
+                                                     conn=source_conn),
+                                             target_conn,
+                                             target_exp_id)
+    finally:
+        source_conn.close()
+        target_conn.close()
+
+
+def _create_exp_if_needed(target_conn: SomeConnection,
+                          exp_name: str, sample_name: str,
+                          fmt_str: str,
+                          start_time: float,
+                          end_time: Union[float, None]) -> int:
+    """
+    Look up in the database whether an experiment already exists and create
+    it if it doesn't. Note that experiments do not have GUIDs, so this method
+    is not guaranteed to work. Matching names and times is the best we can do.
+    """
+
+    time_eq = "=" if end_time is not None else "IS"
+
+    exp_exists_query = f"""
+                       SELECT exp_id
+                       FROM experiments
+                       WHERE name = ?
+                       AND sample_name = ?
+                       AND format_string = "{fmt_str}"
+                       AND start_time = ?
+                       AND end_time {time_eq} ?
+                       """
+    values = (exp_name, sample_name, start_time, end_time)
+    cursor = target_conn.execute(exp_exists_query, values)
+
+    rows = cursor.fetchall()
+
+    if len(rows) > 0:
+        return rows[0]['exp_id']
```

Comment:
You are right, this should be handled. Since runs can always be found via their GUID, I don't think we should raise. I am going to make this a warning.